File: TrafficControl/src/trafficcontrol/Phase.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package trafficcontrol;

/**
 *
 * @author owner
 */
public class Phase {
    
    public Phase(Lane l1, Lane l2) {
        this.L1 = l1;
        this.L2 = l2;
    }
    
    Lane L1;
    Lane L2;
    int TPmax=90;
    int TQx=0; // time till al queue pass
    int TPx=0; //max allowed time for green
    int Qx=0; // not used
    static int lightChangeTime = 0;

    
    public boolean changeLightStatus(){
        L1.lightStatus = !(L1.lightStatus);
        L2.lightStatus = !(L2.lightStatus);
        lightChangeTime = (int) System.currentTimeMillis();
        Qx = getQx();
        TQx = getTQx();
        TPx = getTPx();
        if((L1.lightStatus && L2.lightStatus)==true){return true;}
        else if((L1.lightStatus && L2.lightStatus)==false){return true;}
        
        return false;
    }
    int getQx(){
        if(L1.Q>L2.Q){return L1.Q;}
        else{return L2.Q;}
    }
    int getTQx(){
        L1.computeTQ();
        L2.computeTQ();
        if(L1.TQ>L2.TQ){return L1.TQ;}
        else{return L2.TQ;}
    }
    int getTPx(){
        if(TQx<TPmax){return TQx;}
        else{return TPmax;}
    }
}

File: Traffic control/Phase.py
class Phase(object):
    lightchangetime = 0

    def __init__(self, l1, l2):
        self.L1 = l1
        self.L2 = l2
        self.TPmax = 90
        self.TQx = 0  # time till al queue pass
        self.TPx = 0  # max allowed time for green
        self.Qx = 0  # not used

    def getqx(self):
        if self.L1.Q > self.L2.Q:
            return self.L1.Q
        else:
            return self.L2.Q

    def gettqx(self):
        self.L1.computetq()
        self.L2.computetq()
        if self.L1.TQ > self.L2.TQ:
            return self.L1.TQ
        else:
            return self.L2.TQ

    def gettpx(self):
        if self.TQx < self.TPmax:
            return self.TQx
        else:
            return self.TPmax

    def changelightstatus(self):
        self.L1.lightStatus = not self.L1.lightStatus
        self.L2.lightStatus = not self.L2.lightStatus
        Phase.lightchangetime = self.L1.millis()
        self.Qx = self.getqx()
        self.TQx = self.gettqx()
        self.TPx = self.gettpx()

        if self.L1.lightStatus and self.L2.lightStatus is True:
            return True
        elif self.L1.lightStatus and self.L2.lightStatus is False:
            return True
        return False
